network-config-analyzer/PortSet.py

```diff
@@ -113,178 +123,79 @@ def __init__(self, source_ports=PortSet(), dest_ports=PortSet(), methods=None, p
-            if port_name in res.named_ports:
-                res.named_ports[port_name] -= other.named_ports[port_name]
-
-        res.excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res.excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res.excluded_named_ports:
-                res.excluded_named_ports[port_name] |= other.named_ports[port_name]
-            else:
-                res.excluded_named_ports[port_name] = other.named_ports[port_name]
-
+        res = self.copy()
+        res -= other
         return res
 
     def __iand__(self, other):
-        self.cubes_set &= other.cubes_set
-
-        res_named_ports = dict({})
-        for port_name in self.named_ports:
-            if port_name in other.named_ports:
-                src_interval_res = self.named_ports[port_name] & other.named_ports[port_name]
-                res_named_ports[port_name] = src_interval_res
-        self.named_ports = res_named_ports
-
-        res_excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res_excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.excluded_named_ports:
-            if port_name in res_excluded_named_ports:
-                res_excluded_named_ports[port_name] |= other.excluded_named_ports[port_name]
-            else:
-                res_excluded_named_ports[port_name] = other.excluded_named_ports[port_name]
-        self.excluded_named_ports = res_excluded_named_ports
-
+        assert not self.excluded_named_ports
+        assert not isinstance(other, TcpProperties) or not other.named_ports
+        assert not isinstance(other, TcpProperties) or not other.excluded_named_ports
+        super().__iand__(other)
         return self
 
     def __ior__(self, other):
-        self.cubes_set |= other.cubes_set
-
-        res_named_ports = dict({})
-        for port_name in self.named_ports:
-            res_named_ports[port_name] = self.named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res_named_ports:
-                res_named_ports[port_name] |= other.named_ports[port_name]
-            else:
-                res_named_ports[port_name] = other.named_ports[port_name]
-        self.named_ports = res_named_ports
-
-        res_excluded_named_ports = dict({})
-        for port_name in self.excluded_named_ports:
-            res_excluded_named_ports[port_name] = self.excluded_named_ports[port_name]
-        for port_name in other.named_ports:
-            if port_name in res_excluded_named_ports:
-                res_excluded_named_ports[port_name] -= other.named_ports[port_name]
-        self.excluded_named_ports = res_excluded_named_ports
-
+        assert not isinstance(other, TcpProperties) or not other.excluded_named_ports
+        super().__ior__(other)
```

Comment:
It seems that the 'or' operation between objects with named ports is only used in k8s policy parsing of ports. (and it has no negative ports, and no source ports). should we simplify? add some documentation/assumptions?
(in the calico parsing is not using operators between objects of this type, only creates them).